packages/tracing/src/hubextensions.ts

```diff
@@ -114,10 +117,6 @@ function sample<T extends Transaction>(hub: Hub, transaction: T, samplingContext
     return transaction;
   }
 
-  // at this point we know we're keeping the transaction, whether because of an inherited decision or because it got
-  // lucky with the dice roll
-  transaction.initSpanRecorder(options._experiments?.maxSpans as number);
```

Comment:
The culprit of #3254 was that in https://github.com/getsentry/sentry-javascript/pull/3255/files#diff-f5bf6e0cdf7709e5675fcdc3b4ff254dd68f3c9d1a399c8751e0fa1846fa85dbR54 we returned earlier and skipped initializing a span recorder.
Worked fine for forcing `sampled: false`, but clearly not the intended behavior for `sampled: true`.